app/lib/searching.rb

```diff
@@ -55,8 +75,85 @@ def search_left_list_for_link(link)
     end
 
     def score_difference(link, position)
-      return (search_left_list_for_link(link) - position).to_s if search_left_list_for_link(link)
-      "++++"
+      search_left_list_for_link(link) ? (search_left_list_for_link(link) - position).to_s : "++++"
+    end
+  end
+
+  class Result
+    require 'uri'
+    delegate :[], to: :@info
+    def initialize(info)
+      @info = info
+    end
+
+    def enhanced_results(enhanced_fields)
+      return_hash = {}
+      enhanced_fields.each do |field|
+        next unless @info[field].present?
+        return_hash[field.titleize] = @info[field].map { |t| link_title_array(t, field) }
+      end
+      return_hash
+    end
+
+    def get_head_info_list(fields)
+      head_info_list = [@info['format'].humanize, date_format(@info['public_timestamp'])]
+      head_info_list << historical_or_current(@info['is_historic']) if fields.include?("is_historic")
+      head_info_list << "Popularity: #{@info['popularity']}" if fields.include?("popularity")
+      head_info_list
+    end
+
+    def link
+      format_link(@info['link'])
+    end
+
+    def name
+      make_readable(@info['link']).strip
+    end
+
+    def second_head(fields)
+      (fields.include?("people") ? people : []) + (fields.include?("organisations") ? organisations : [])
+    end
+
+  private
+
+    def date_format(date)
+      DateTime.parse(date).strftime("%B %Y") unless date.nil?
+    end
+
+    def format_link(link, extra = "")
+      return link if link == nil || link.start_with?("https://", "http://")
+      return "https://#{link}" if link.start_with?("www.")
+      "https://gov.uk" + extra + link
+    end
+
+    def historical_or_current(check)
+      return if check.nil?
+      check ? "Historical" : "Current"
+    end
+
+    def link_title_array(link, field)
```

Comment:
This is a badly named method as it has nothing to do with arrays

Reply:
has now been renamed